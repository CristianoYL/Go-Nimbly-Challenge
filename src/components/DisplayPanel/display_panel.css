.display-panel {
  color: var(--color-accent);
  background-color: var(--color-primary);
  height: 100%;
}

.display-text {
  height: 100%;
  display: flex;
  flex-flow: column nowrap;
  justify-content: center;
  align-items: flex-start;
  margin: 0 7rem;
}

h3 {
  font-size: 4rem;
  font-weight: 700;
}

p {
  font-size: 125%;
}
