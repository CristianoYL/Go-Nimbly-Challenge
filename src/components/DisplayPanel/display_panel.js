import React from 'react';
import PropTypes from 'prop-types';
import './display_panel.css';

const DisplayPanel = ({ children }) => (
  <div className="display-panel">
    <div className="display-text">{children}</div>
  </div>
);

DisplayPanel.defaultProps = {
  children: null,
};

DisplayPanel.propTypes = {
  children: PropTypes.node,
};

export default DisplayPanel;
