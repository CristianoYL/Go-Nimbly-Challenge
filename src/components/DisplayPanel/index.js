import DisplayPanel from './display_panel';

export default DisplayPanel;
