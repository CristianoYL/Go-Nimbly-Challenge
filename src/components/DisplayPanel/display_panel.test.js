import React from 'react';
import renderer from 'react-test-renderer';
import DisplayPanel from './display_panel';

test('DisplayPanel without children', () => {
  const component = renderer.create(<DisplayPanel />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('DisplayPanel with one child', () => {
  const component = renderer.create(
    <DisplayPanel>
      <div />
    </DisplayPanel>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('DisplayPanel with two children', () => {
  const component = renderer.create(
    <DisplayPanel>
      <h1>Test</h1>
      <div />
    </DisplayPanel>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('DisplayPanel with nested children', () => {
  const component = renderer.create(
    <DisplayPanel>
      <div>
        <h1>Test</h1>
      </div>
    </DisplayPanel>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
