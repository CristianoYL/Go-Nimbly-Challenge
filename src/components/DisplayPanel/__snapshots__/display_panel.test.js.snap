// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplayPanel with nested children 1`] = `
<div
  className="display-panel"
>
  <div
    className="display-text"
  >
    <div>
      <h1>
        Test
      </h1>
    </div>
  </div>
</div>
`;

exports[`DisplayPanel with one child 1`] = `
<div
  className="display-panel"
>
  <div
    className="display-text"
  >
    <div />
  </div>
</div>
`;

exports[`DisplayPanel with two children 1`] = `
<div
  className="display-panel"
>
  <div
    className="display-text"
  >
    <h1>
      Test
    </h1>
    <div />
  </div>
</div>
`;

exports[`DisplayPanel without children 1`] = `
<div
  className="display-panel"
>
  <div
    className="display-text"
  />
</div>
`;
